src/analysis/fourier/fourier_transform.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import analysis.special_functions.complex.circle
+import measure_theory.group.integration
+
+/-!
+# The Fourier transform
+
+We set up the Fourier transform for complex-valued functions on finite-dimensional spaces.
+
+## Design choices
+
+We define Fourier transforms in the following context:
+* `𝕜` is a commutative ring.
+* `V` and `W` are `𝕜`-modules.
+* `e` is a unitary additive character of `𝕜`, i.e. a homomorphism `(multiplicative 𝕜) →* circle`.
+* `μ` is a measure on `V`.
+* `L` is a `𝕜`-bilinear form `V × W → 𝕜`.
+* `E` is a complete normed `ℂ`-vector space.
+
+With these definitions, the Fourier transform is a map from functions `V → E` to
+functions `W → E`, defined by sending `f` to
+
+`λ w, ∫ v in V, e [-L v w] • f v ∂μ`,
+
+where `e [x]` is notational sugar for `(e (multiplicative.of_add x) : ℂ)` (available in locale
+`fourier_transform`).
+
+The most familiar case, of course, is
+* `𝕜 = V = W = ℝ`
+* `e = fourier_char` (defined below), i.e. the character `λ x, exp (2 * π * I * x)`
+* `μ = volume`
+* `L = linear_map.mul ℝ ℝ`
+
+The notation `𝓕` is available in the locale `fourier_transform` as a shortcut for this case.
+However, we set things up much more generally (number theorists care about the case when `𝕜` is an
+adele ring, for instance).
+
+## Main results
+
+At present the only nontrivial lemma we prove is `continuous_fourier_transform`, stating that the
+Fourier transform of an integrable function is continuous (under mild assumptions).
+-/
+
+noncomputable theory
+
+local notation `𝕊` := circle
+
+open measure_theory filter
+
+open_locale topology
+
+-- To avoid messing around with multiplicative vs. additive characters, we make a notation.
+localized "notation e `[` x `]` := (e (multiplicative.of_add x) : ℂ)" in fourier_transform
+
```

Comment:
I expect that most analysts will not really care about the general version, and will mostly use specialized versions to the real line or to inner product spaces. A good solution for this would be to have the general theorems in some namespace, and then specialized versions of the statements in the `real` namespace or in the `inner_product` namespace, say. So I think we should put most results of this file in a namespace. `general_fourier`? (not very good, I think, but I don't have a better idea).

Reply:
I thought mathlib's general convention is to do everything in as much generality as possible, rather than trying to decide in advance which cases are "enough for most use cases" and hide more general ones out of the way. (The existing treatment of convolutions, on which my fourier-integral code is heavily modelled, is an example.)

Reply:
I definitely think your approach is the right one, and I am not asking for any change regarding generality here. I only say that namespacing the construction is probably a good idea towards a smooth future.

Reply:
OK, here's a version with namespaces as follows:
- `vector_fourier`: the full generality as before
- `fourier`: the case `V = W = 𝕜` and `L = mul`
- `real`: the case `V = W = 𝕜 = ℝ`, `L = mul` and `e = fourier_char`

What do you think?